src/RouterWrapper.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState } from "react";
+import ChrisAPIClient from "./api/chrisapiclient";
+import { AppContext } from "./context/context";
+import { Types } from "./context/actions/types";
+import { IUserState } from "./context/reducers/userReducer";
+import Client, { User } from "@fnndsc/chrisapi";
+import { useLocation, useHistory } from 'react-router-dom'
+
+/**
+ * Allows user to bypass Login page if already authenticated by checking AUTH_TOKEN
+ * in session storage
+ */
+const RouterWrapper: React.FC = ({ children }) => {
+  const { dispatch } = React.useContext(AppContext);
+  const [renderChildren, setRenderChildren] = useState(false);
+  const location = useLocation();
+  const history = useHistory();
+
+  useEffect(() => {
+    const token = window.sessionStorage.getItem("AUTH_TOKEN");
+    (async () => {
+      if (!!token) {
+        const client: Client = ChrisAPIClient.getClient();
+        const res: User = await client.getUser();
+        const user: IUserState = res?.data;
+        dispatch({
+          type: Types.Login_update,
+          payload: user
+        });
+        
+        // If on Login page and already authenticated, go to Dashboard
+        if (location.pathname === "/login") {
+          history.push('/');
```

Comment:
How come you need to route them from the RouterWRapper? How was previous the user routed to the Dashboard previously?

Reply:
Previously, if auth token already existed, a user would still need to manually log into COVIDNet. The routing above redirects a user to the Dashboard if a token already exists.